src/native/corehost/corehost.cpp

```diff
@@ -74,8 +87,8 @@ bool is_exe_enabled_for_execution(pal::string_t* app_dll)
     size_t hi_len = (sizeof(hi_part) / sizeof(hi_part[0])) - 1;
     size_t lo_len = (sizeof(lo_part) / sizeof(lo_part[0])) - 1;
     if (binding.size() >= (hi_len + lo_len)
-        && binding.compare(0, hi_len, &hi_part[0]) == 0
-        && binding.compare(hi_len, lo_len, &lo_part[0]) == 0)
+        && compare_memory_nooptimization(binding.c_str(), hi_part, hi_len)
+        && compare_memory_nooptimization(binding.substr(hi_len).c_str(), lo_part, lo_len))
```

Comment:
Consider:
```c
        && compare_memory_nooptimization(&embed[0], hi_part, hi_len)
        && compare_memory_nooptimization(&embed[hi_len], lo_part, lo_len)
```
Also, the `std::string binding` in this function can be removed by calling `strlen` on `embed`.